transcribers/mqtt.py

```diff
@@ -131,28 +133,53 @@ def _mqtt_to_ipal(self, request, pkt):
             data=data,
         )
 
+        # depending on the message type, add the message to the request queue or match it to a request in the queue
+        ipal_message._add_to_request_queue = (type in MQTTProtocol.REQUEST_COMMANDS)
+        ipal_message._match_to_requests = (type in MQTTProtocol.RESPONSE_COMMANDS)
+
+        return ipal_message
+
     def match_response(self, requests, response):
         remove_from_queue = []
 
-        # preemptively reduce the amount of checks we need to do
-        match_types = MQTTProtocol.command_response(response)
-        if len(match_types) == 0:
-            return []
-
-        # then only check relevant types
         for request in requests:
-            if request.type in match_types:
```

Comment:
Instead of removing this, you should leverage the performance gain when looping through the requests.

Reply:
I have included this in the latest commit.